tests/test_grid_sampling.py

```diff
@@ -21,6 +22,40 @@ def grid(xdim=200, ydim=100):
                           depth, time)
 
 
+@pytest.fixture
+def grid_geographic(grid):
+    # Apply inverse conversion (from m to degree)
+    grid.U.data *= 1000. * 1.852 * 60.
+    grid.V.data *= 1000. * 1.852 * 60.
+    grid.U.units = Geographic()
+    grid.V.units = Geographic()
+    return grid
+
+
+@pytest.fixture
+def grid_geographic_polar(grid_geographic):
+    grid = grid_geographic
+    for i, y in enumerate(grid.U.lat):
+        # Apply inverse conversion for each latitude value
+        grid.U.data[0, i, :] *= cos(y * pi / 180)
+    grid.U.units = GeographicPolar()
+    return grid
```

Comment:
Note that in virtually all cases, U and V will have the same grid layout. It is only the lon and lat dimensions that are different in geographic_polar